File: python/19_counting_sundays.py
import datetime

def count_sundays() -> int:
    count: int = 0
    # use datetime object to check day of week for the first of every month from 1901-01-01 to 2000-12-31
    for year in range(1901, 2001):
        for month in range(1, 13):
            if datetime.date(year, month, 1).weekday() == 6:
                count += 1
    return count

if __name__ == "__main__":
    print(count_sundays())

File: c#/_19_CountingSundays.cs
using System;

public class _19_CountingSundays{
	public static int Euler19(){
		DateTime date = new DateTime(1901, 1, 1);
		int count = 0;
		while (date.Year != 2001){
			if (date.DayOfWeek.Equals(DayOfWeek.Sunday)) count++;
			date = date.AddMonths(1);
		}
		return count;
	}
	
	public static void Main(){
		Console.WriteLine(Euler19());
	}
}
